treeherder/webapp/api/push.py

```diff
@@ -87,7 +87,7 @@ def list(self, request, project):
                 # revision can be either the revision of the push itself, or
                 # any of the commits it refers to
                 revision_field = 'revision__startswith' if len(value) < 40 else 'revision'
-                filter_kwargs = {revision_field: value}
+                filter_kwargs = {revision_field: value, 'repository': repository}
```

Comment:
`pushes` is filtered by `repository` higher up in the file, so this line can stay as is I think?

Reply:
Shoot, totally true.  OK, fixed.